src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php

```diff
@@ -46,7 +47,11 @@ public function mapDataToForms($data, $forms)
             $config = $form->getConfig();
 
             if (!$empty && null !== $propertyPath && $config->getMapped()) {
-                $form->setData($this->propertyAccessor->getValue($data, $propertyPath));
+                try {
+                    $form->setData($this->propertyAccessor->getValue($data, $propertyPath));
+                } catch (AccessException $e) {
```

Comment:
I think we should either target 3.4 and/or catch the new `UninitializedPropertyException`.

Reply:
I changed the target branch to 3.4.
I'm not sure if catching `UninitializedPropertyException` provides an advantage over the broader `AccessException` here. Is there a scenario in which catching the other possible AccessExceptions is harmful in the light of a flexible implementation?

Reply:
The advantage is that the `AccessException` can be used internally by the PropertyAccess component for a "broader" usage, which should not be covered here.

Reply:
And yes it could lead to bug because the form should break as misconfigured so the dev can fix it, because a property is not readable/writable as it should.
That's why I think we really should consider master instead and consider this a new feature unlocked by the new exception (that was my original intent, but I'm glad you opened that PR :)).

Reply:
Another solution for targeting 3.4 would be to check the exception message to ensure the catch is legit.

Reply:
How do you think about this?
```php
if (!$e instanceof UninitializedPropertyException
    // For versions without UninitializedPropertyException check the exception message
    && (\class_exists(UninitializedPropertyException::class) || false === \strpos($e->getMessage(), 'You should initialize it'))
) {
    throw $e;
}
```

Reply:
We would only need to check the message in 3.4 and then when merging branches up, use the type check in master instead. No need for more complexity here in your patch.

Reply:
For 3.4 the type is never available with the current dependency constraint.
For 4.4, 5.0 and master it might be available.
I think we need to keep both checks in place until that dependency is bumped.